Gizra/og
# Fix error when resolving the entity on entity.node.revision route

When an entity's render array has 'og_membership_state' cache
context, then OG's route resolvers error out when viewing that
node's earlier revisions on the entity.node.revision route.

This happens because $route->getParameter('node') returns a string
ID of the entity instead of the fully loaded entity.

## Reproduction steps for testing

I tested this on Gizra/drupal-starter which comes already with pluggable entity view builder module.

1. Start ddev and install og `ddev start && ddev composer require drupal/og:1.x-dev`
1. Enable og and og_ui `ddev . drush --no-interaction en og og_ui`
1. Go to Structure -> Content types and
   1. Set Basic Page to be a Group
   1. Set Article to be Basic Page's Group content
1. Create a basic page node : Test group
1. Create an article node : Test group content
1. Edit the article and create a new revision with some changes.
1. Add `og_membership_state` cache context on the article's render array. To do this we edit `Drupal\server_general\Plugin\EntityViewBuilder\NodeArticle` class's `buildFull` method. Add the following line before the `return $build;` line:
   ```php
   CacheableMetadata::createFromRenderArray($build)->addCacheContexts(['og_membership_state'])->applyTo($build);
   ``` 
   So the whole function becomes:

   ```php
   public function buildFull(array $build, NodeInterface $entity) {
     $build['header'] = $this->buildHeroHeader($entity);
     $build['tags'] = $this->buildContentTags($entity);
     $build['body'] = $this->buildBody($entity);
     CacheableMetadata::createFromRenderArray($build)->addCacheContexts(['og_membership_state'])->applyTo($build);
     return $build;
   }
   ```

   (Don't forget to import the CacheableMetadata class :) )
1. Open the article's revision we created earlier. If this was a fresh install it's probably going to be at `/node/2/revisions/2/view` path.

Expected result:

Opens revision

Actual result:

`Error: Call to a member function getEntityTypeId() on string in Drupal\og\Plugin\OgGroupResolver\RouteGroupResolver->resolve()`
